### PR TITLE
Fix bugs + a new feature 

### LINKED ISSUE #125: fix: render "null" rather than 1 for the winner in SingleDistrict in quotient columns

### PR DESCRIPTION
**Pull request by oysor**
# Description

New file created: SmartNumericInputWithLabel.tsx

### CODE REVIEW

File: src/Layout/Presentation/SingleDistrict/SingleDistrict.tsx [hunk 1]
```diff
@@ -135,6 +135,9 @@ export class SingleDistrict extends React.Component<SingleDistrictProps, {}> {
                                 if (row.original.partyCode === vulnerableVotes.partyCode) {
                                     return <div className="has-background-dark has-text-white">{row.value}</div>;
                                 }
+                                if (row.value === 1) {
+                                    return "null";
+                                }
```
Comment (simendag) [inline src/Layout/Presentation/SingleDistrict/SingleDistrict.tsx:140]: This is incorrect. You need to check for the winning party, not for the value.

File: src/common/SmartNumericInput/SmartNumericInput.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ export interface SmartNumericInputProps {
     hidden?: boolean;
 }
 
-export class SmartNumericInput extends React.Component<SmartNumericInputProps, {}> {
+export class SmartNumericInput<T extends SmartNumericInputProps> extends React.Component<T, {}> {
```
Comment (simendag) [inline src/common/SmartNumericInput/SmartNumericInput.tsx:18]: Why does this component now have a type parameter?

File: src/common/SmartNumericInputWithLabel/SmartNumericInputWithLabel.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+import { SmartNumericInputProps, SmartNumericInput } from "../SmartNumericInput";
+
+export interface SmartNumericInputWithLabelProps extends SmartNumericInputProps {
+    label?: string;
+}
+
+export class SmartNumericInputWithLabel extends SmartNumericInput<SmartNumericInputWithLabelProps> {
```
Comment (simendag) [inline src/common/SmartNumericInputWithLabel/SmartNumericInputWithLabel.tsx:8]: It doesn't work to use SmartNumericInputWithLabelProps?

File: src/Layout/Presentation/SingleDistrict/SingleDistrict.tsx (hunk 1 above)
Comment (oysor) [reply src/Layout/Presentation/SingleDistrict/SingleDistrict.tsx:140]:
```
     if (row.original.partyCode === vulnerableVotes.winner.partyCode) {
              return "null";
         }
```

File: src/common/SmartNumericInputWithLabel/SmartNumericInputWithLabel.tsx (hunk 3 above)
Comment (oysor) [reply src/common/SmartNumericInputWithLabel/SmartNumericInputWithLabel.tsx:8]: Not if I want to use the methods in SmartNumericInput also.

File: src/Layout/Presentation/SingleDistrict/SingleDistrict.tsx (hunk 1 above)
Comment (simendag) [reply src/Layout/Presentation/SingleDistrict/SingleDistrict.tsx:140]: I think this should work. 👍

File: src/common/SmartNumericInput/SmartNumericInput.tsx (hunk 2 above)
Comment (oysor) [reply src/common/SmartNumericInput/SmartNumericInput.tsx:18]: In order to add additional props and get use of the methods by extending it.

File: src/common/SmartNumericInputWithLabel/SmartNumericInputWithLabel.tsx (hunk 3 above)
Comment (simendag) [reply src/common/SmartNumericInputWithLabel/SmartNumericInputWithLabel.tsx:8]: You can probably extend the class of the component, and implement the interface for the property you need.

Comment (oysor) [reply src/common/SmartNumericInputWithLabel/SmartNumericInputWithLabel.tsx:8]: If I do 
`export class SmartNumericInputWithLabel extends SmartNumericInput implements SmartNumericInputWithLabelProps {`

I'll get this error:
`
Property 'label' does not exist on type 'Readonly<SmartNumericInputProps> & Readonly<{ children?: ReactNode; }>'.
`
So the interface of SmartNumericInput is not a friend of interface SmartNumericInputWithLabelProps (even though the latter interface extends the former)

If I do 
`export class SmartNumericInputWithLabel extends SmartNumericInput implements React.Component<SmartNumericInputWithLabelProps, {}> {`
```
Class 'SmartNumericInputWithLabel' incorrectly implements class 'Component<SmartNumericInputWithLabelProps, {}, any>'. Did you mean to extend 'Component<SmartNumericInputWithLabelProps, {}, any>' and inherit its members as a subclass?
  Types of property 'props' are incompatible.
    Type 'Readonly<SmartNumericInputProps> & Readonly<{ children?: ReactNode; }>' is not assignable to type 'Readonly<SmartNumericInputWithLabelProps> & Readonly<{ children?: ReactNode; }>'.
      Type 'Readonly<SmartNumericInputProps> & Readonly<{ children?: ReactNode; }>' is not assignable to type 'Readonly<SmartNumericInputWithLabelProps>'.
        Type 'Readonly<SmartNumericInputProps> & Readonly<{ children?: ReactNode; }>' is missing the following properties from type 'Readonly<SmartNumericInputWithLabelProps>': context, setState, forceUpdate, render, and 3 more.
```

**Comment by simendag** [comment]
In retrospect I was wrong to suggest using inheritance here 👀. It might be easier to use a functional approach if we want to refrain from "touching" other components, and rely on composition. If you are unfamiliar with higher order components we can schedule a meeting tomorrow and run through it @oysor -- alternatively you can test the original solution you suggested thoroughly (any component you affect, you have to test thoroughly).

**Pull request merged**